packages/adp-tooling/src/preview/routes-handler.ts

```diff
@@ -0,0 +1,202 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import sanitize from 'sanitize-filename';
+import type NodeCache from 'node-cache';
+import type { ReaderCollection } from '@ui5/fs';
+import type { ToolsLogger } from '@sap-ux/logger';
+import type { NextFunction, Request, Response } from 'express';
+
+import { FolderNames, TemplateFileName, HttpStatusCodes } from '../types';
+
+interface WriteFragmentBody {
+    fragmentName: string;
+}
+
+/**
+ * @description Handles API Routes
+ */
+export default class RoutesHandler {
+    /**
+     * Constructor taking project as input.
+     *
+     * @param project Reference to the root of the project
+     * @param logger Logger instance
+     * @param cache Caching instance
+     */
+    constructor(
+        private readonly project: ReaderCollection,
+        private readonly logger: ToolsLogger,
+        private cache: NodeCache
+    ) {}
+
+    /**
+     * Generic caching function, caches the results of a callback
+     *
+     * @param key Cache key - unique identifier
+     * @param cb Callback used to return data
+     * @param ttlSeconds Time-to-live for cached item
+     * @template T
+     * @returns {T} Type or Interface of the return data
+     */
+    private withCache<T>(key: string, cb: () => T, ttlSeconds: number = 60): T {
+        const cachedData = this.cache.get<T>(key);
+
+        if (cachedData !== undefined) {
+            return cachedData;
+        } else {
+            const data = cb();
+            this.cache.set(key, data, ttlSeconds);
+            return data;
+        }
+    }
+
+    /**
+     * Handler for retrieving XML Fragments for templates used for Dialogs
+     *
+     * @param req Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleGetXMLFragmentByName = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Can you explain for what we need this function? Why do you need expose the any of the fragments from the templates folder?

Reply:
Because as you can see I've removed the dialog from `dialogs/fragment.ts` and put it inside an XML file and it is now following the best practices of UI5. Now we have a UI folder with all the fragments, there will be dialog also for a controller extension and maybe other stuff later. We instantiate it in the fragment.ts file. And then it is looking for the fragment at /resources/adp/fragments/<name>.

Reply:
I am sorry, but no, "I cannot see". As far as I can see, you are working on a rather simple create and list all functionality. You client code needs a list of fragments in the workspace (simple `byGlob` for `.fragment.xml` files), and then your client either allows assigned one of these files to a change (no backend request needed) or to create a new fragment. For creating the new fragement, you client never needs to see the template, you just create a new file by copying the template when your post fragment request is triggered.

So, again, why do you need a function that exposes files from the template folder?

Reply:
Wait a second. Why is the `add-fragment.fragment.xml` in the `templates` folder and not in `src/preview/client` (maybe a subfolder)? This file is static client code and shouldn't be templated.